src/modules/job-list/idsync.c

```diff
@@ -174,26 +185,31 @@ void idsync_check_id_valid_cleanup (struct idsync_ctx *isctx,
 static int idsync_add_waiter (struct idsync_ctx *isctx,
                               struct idsync_data *isd)
 {
-    zlistx_t *list_isd;
+    struct idsync_wait_list *iwl = NULL;
 
     /* isctx->waits holds lists of ids waiting on, b/c multiple callers
      * could wait on same id */
-    if (!(list_isd = zhashx_lookup (isctx->waits, &isd->id))) {
-        if (!(list_isd = zlistx_new ()))
+    if (!(iwl = zhashx_lookup (isctx->waits, &isd->id))) {
+        iwl = calloc (1, sizeof (*iwl));
+        if (!iwl)
+            goto enomem;
+
+        if (!(iwl->l = zlistx_new ()))
             goto enomem;
-        zlistx_set_destructor (list_isd, idsync_data_destroy_wrapper);
+        zlistx_set_destructor (iwl->l, idsync_data_destroy_wrapper);
+        iwl->id = isd->id;
 
-        if (zhashx_insert (isctx->waits, &isd->id, list_isd) < 0)
+        if (zhashx_insert (isctx->waits, &iwl->id, iwl) < 0)
             goto enomem;
     }
 
-    if (!zlistx_add_end (list_isd, isd))
+    if (!zlistx_add_end (iwl->l, isd))
         goto enomem;
 
     return 0;
 
 enomem:
-    zlistx_destroy (&list_isd);
+    idsync_wait_list_destroy ((void **)&iwl);
```

Comment:
cast to `(void **)` is unnecessary here I think.

Reply:
i get a error/warning with it :-(

Reply:
Oh, ok, my bad then!